pystac/extensions/version.py

```diff
@@ -0,0 +1,170 @@
+"""Implement the version extension.
+
+https://github.com/radiantearth/stac-spec/tree/dev/extensions/version
+"""
+
+import pystac
+from pystac import collection
+from pystac import Extensions
+from pystac import item
+from pystac import link
+from pystac.extensions import base
+
+LATEST_VERSION = 'latest-version'
+PREDECESSOR_VERSION = 'predecessor-version'
+SUCCESSOR_VERSION = 'successor-version'
+
+# Media type for links.
+MEDIA_TYPE = 'application/json'
+
+
+class VersionItemExt(base.ItemExtension):
+    """Add an asset version string to a STAC Item."""
+    def __init__(self, an_item):
+        self.item = an_item
+
+    def apply(self, version, deprecated=None, latest=None, predecessor=None, successor=None):
+        self.version = version
+        if deprecated is not None:
+            self.deprecated = deprecated
+        if latest:
+            self.latest_link = latest
+        if predecessor:
+            self.predecessor_link = predecessor
+        if successor:
+            self.successor_link = successor
+
+    @property
+    def version(self):
+        return self.item.properties.get('version')
+
+    @version.setter
+    def version(self, v):
+        self.item.properties['version'] = v
+
+    @property
+    def deprecated(self):
+        return bool(self.item.properties.get('deprecated'))
+
+    @deprecated.setter
+    def deprecated(self, v):
+        if not isinstance(v, bool):
+            raise pystac.STACError('deprecated must be a bool')
+        self.item.properties['deprecated'] = v
+
+    @property
+    def latest_link(self):
```

Comment:
I find the `_link` part of the method names and the differing setter values and return values a bit confusing. We aren't passing links in the setter, but the getter returns links... IMO it'd be better to have these property methods as `latest`, `predecessor`, and `successor`, similarly to how they appear in the `apply` parameter list, and then have them both get and set the target object instead of the link.

Reply:
Agreed.  Done in b35cd64.

I will also remove `_VERSION` from the constants that hold the field names  So `version.LATEST_VERSION` → `version.LATEST`.  `version.LATEST_VERSION` is painfully redundant.

Reply:
Sounds good.

Looks like that commit still returns `Link` objects from the getters. To make things consistent, I think we should return the Item or Collection from those getters (similarly to how that is the value passed in), e.g.:

```python
@property
    def latest(self):
        links = self.item.get_links(LATEST_VERSION)
        if links:
            return links[0].resolve_stac_object(root=self.item.get_root()).target
```

the `resolve_stac_object` call on the link will make sure that the target is the relevant PySTAC object and not a link HREF. The `root` argument is a bit clunky, and I think the API needs to change around the link resolution code, but for now that should do the right thing.

Reply:
Actually, potentially cleaner to use this method: https://github.com/stac-utils/pystac/blob/9944c5178b97333ceb5ff74b0d8a516acce5f22d/pystac/stac_object.py#L346

a la 

https://github.com/stac-utils/pystac/blob/develop/pystac/extensions/label.py#L261

Reply:
Done in https://github.com/stac-utils/pystac/pull/193/commits/d408d2e24b91b9d444b6f1285363c017e13fd7a5.  Also make VERSION and DEPRECATED constants